### PR TITLE
REV-54-connect listing page with backend

### PR DESCRIPTION
**Pull request by IbrahimAljabrST**
### Jira URL

https://optimumpartners.atlassian.net/browse/REV-54

### Description

connect listing page with backend

### Dependencies in Other PRs,Repo

https://github.com/abukhadraop-or/reverifi-api/pull/17

### Testing Instructions

open listing page with listing/id route

### CODE REVIEW

File: src/components/listing-page/Details/index.jsx [hunk 1]
```diff
@@ -11,21 +11,64 @@ import {
...
- * @param {Object} props         The component props.
- * @param {Object} props.details Details section data.
+
+ * @param {Object} details Details section data.
  *
  * @return {JSX.Element}
  */
 function Details({ details }) {
+  const [listingDetails, setListingDetails] = useState([]);
+  const [listingSize, setListingSize] = useState([]);
```
Comment (AhmadAlShwaiki) [inline src/components/listing-page/Details/index.jsx:21]: Do we really need a second state here?

File: src/components/listing-page/Features/index.jsx [hunk 2]
```diff
@@ -17,44 +13,29 @@ import {
  *
  * @return {JSX.Element}
  */
-function Features({ data }) {
-  const { features, icons } = data;
-
+// eslint-disable-next-line react/prop-types
+function Features({ features }) {
+  if (features.length === 0) return null;
```
Comment (AhmadAlShwaiki) [inline src/components/listing-page/Features/index.jsx:18]:
```suggestion
  if (!features.length) return null;
```

File: src/components/listing-page/Features/index.jsx [hunk 3]
```diff
@@ -17,44 +13,29 @@ import {
...
-            features[key] && (
-              <FeaturesItem key={key}>
-                <Image src={icons[index]} alt={key} />
-                <Label>{key}</Label>
-              </FeaturesItem>
-            )
-        )}
+        {data.map((key) => (
+          <FeaturesItem
+            key={key.feature}
```
Comment (AhmadAlShwaiki) [inline src/components/listing-page/Features/index.jsx:25]: Please stick to object destructing

File: src/components/shared/Navbar/navbar.styles.js [hunk 4]
```diff
@@ -47,6 +47,9 @@ export const NavItemsContainer = styled.div`
 export const NavLinksContainer = styled.div`
   display: none;
   font-size: 0.9rem;
+  span {
+    font-weight: 800;
```
Comment (AhmadAlShwaiki) [inline src/components/shared/Navbar/navbar.styles.js:51]:
```suggestion
    font-weight: bold;
```

File: src/components/shared/Navbar/navbar.styles.js [hunk 5]
```diff
@@ -123,10 +126,12 @@ export const Logo = styled(RawLogo)`
 export const SignInButton = styled(Button)`
   background-color: transparent;
   font-size: 0.9rem;
+  font-weight: 500;
```
Comment (AhmadAlShwaiki) [inline src/components/shared/Navbar/navbar.styles.js:129]: Can't see why these values are different, please unite all css values where possible

File: src/components/shared/Navbar/navbar.styles.js [hunk 6]
```diff
@@ -123,10 +126,12 @@ export const Logo = styled(RawLogo)`
 export const SignInButton = styled(Button)`
   background-color: transparent;
   font-size: 0.9rem;
```
Comment (AhmadAlShwaiki) [inline src/components/shared/Navbar/navbar.styles.js:128]: Please adhere to css styling best practices
```suggestion
  font-size: 1rem;
```

File: src/pages/Listing/ListingPage.jsx [hunk 7]
```diff
@@ -29,68 +20,33 @@ const {
...
+  if (listingDetails.id) {
+    return (
+      <ShowModalProvider>
+        <Offer address={listingDetails.address} price={listingDetails.price} />
+        <ListingPageSlider images={listingDetails.images} />
+        <Overview listing={listingDetails} />
+        <Details details={listingDetails} />
+        <Features features={listingDetails.features} />
+        <Location />
+        {/* <SimilarListings similarListings={similarListings} /> */}
```
Comment (AhmadAlShwaiki) [inline src/pages/Listing/ListingPage.jsx:44]: Please remove any commented lines of code

File: src/routes.js [hunk 8]
```diff
@@ -57,7 +57,7 @@ const routes = [
   {
     component: lazy(() => import('pages/Listing/ListingPage')),
     label: 'listing',
-    path: '/listing',
+    path: '/listing/:id',
```
Comment (AhmadAlShwaiki) [inline src/routes.js:60]: Why do we have two listing pages?

File: src/components/shared/Navbar/navbar.styles.js (hunk 5 above)
Comment (IbrahimAljabrST) [reply src/components/shared/Navbar/navbar.styles.js:129]: QA want it like this

File: src/components/shared/Navbar/navbar.styles.js (hunk 6 above)
Comment (IbrahimAljabrST) [reply src/components/shared/Navbar/navbar.styles.js:128]: QA want it like this

**Review by AhmadAlShwaiki — APPROVED** [review]

**Pull request merged**